Nightly Build: Many Warnings in StructuralMechanics

from the nightly build of CLANG:
~~~
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::VonMisesYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::ModifiedMohrCoulombYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::TrescaYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::DruckerPragerYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::RankineYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::VonMisesPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::ModifiedMohrCoulombPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::DruckerPragerPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:175:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::Has' hides overloaded virtual functions [-Woverloaded-virtual]
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.h:213:10: warning: 'Kratos::GenericSmallStrainHighCycleFatigueLaw<Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::TrescaPlasticPotential<6> > > >::GetValue' hides overloaded virtual functions [-Woverloaded-virtual]
~~~

can you guys please have a look?
Thanks

--- a/applications/StructuralMechanicsApplication/custom_advanced_constitutive/generic_small_strain_high_cycle_fatigue_law.h
+++ b/applications/StructuralMechanicsApplication/custom_advanced_constitutive/generic_small_strain_high_cycle_fatigue_law.h
@@ -170,6 +170,7 @@ public:
      * @param rThisVariable the variable to be checked for
      * @return true if the variable is defined in the constitutive law
      */
+    using ConstitutiveLaw::Has;
     bool Has(const Variable<int>& rThisVariable) override;
 
     /**
@@ -188,6 +189,7 @@ public:
      * @param rVariable the variable to be returned
      * @param rValue new value of the specified variable
      */
+    using ConstitutiveLaw::SetValue;
     void SetValue(
         const Variable<int>& rThisVariable,
         const int& rValue);
@@ -208,6 +210,7 @@ public:
      * @param rValue a reference to the returned value
      * @return rValue output: the value of the specified variable
      */
+    using ConstitutiveLaw::GetValue;
     int& GetValue(
         const Variable<int>& rThisVariable,
         int& rValue) override;

--- a/applications/StructuralMechanicsApplication/custom_advanced_constitutive/small_strain_j2_plasticity_3d.h
+++ b/applications/StructuralMechanicsApplication/custom_advanced_constitutive/small_strain_j2_plasticity_3d.h
@@ -328,6 +328,7 @@ protected:
      * @param rPlasticStrain
      * @param rAccumulatedPlasticStrain
      */
+    using ConstitutiveLaw::CalculateStressResponse;
     virtual void CalculateStressResponse(ConstitutiveLaw::Parameters& rValues,
                                  Vector& rPlasticStrain,
                                  double& rAccumulatedPlasticStrain );

